base/strings/annotated.jl

```diff
@@ -386,3 +386,52 @@ annotations(s::SubString{<:AnnotatedString}, pos::UnitRange{<:Integer}) =
 Get all annotations of `chr`.
 """
 annotations(c::AnnotatedChar) = c.annotations
+
+## AnnotatedIOBuffer
+
+struct AnnotatedIOBuffer <: IO
```

Comment:
Maybe just call this `AnnotatedBufffer`? It's missing seek for really being an `IOBuffer`, and `read` is `take!`

Reply:
Could do. We could also forward those functions onto the underlying `IOBuffer`.

Reply:
This is still to be done?